packages/itwinui-react/src/core/Tree/Tree.tsx

```diff
@@ -380,26 +432,74 @@ const VirtualizedTree = React.forwardRef(
     }: VirtualizedTreeProps<T>,
     ref: React.ForwardedRef<HTMLUListElement>,
   ) => {
-    const { outerProps, innerProps, visibleChildren } = useVirtualization({
-      itemsLength: flatNodesList.length,
-      itemRenderer: itemRenderer,
-      scrollToIndex,
+    const parentRef = React.useRef<HTMLDivElement | null>(null);
+    const getScrollableElement = (
+      element: HTMLElement | null,
+      ownerDocument: Document = document,
+    ): HTMLElement => {
+      if (!element || element === ownerDocument.body) {
+        return ownerDocument.body;
+      }
+
+      return isElementScrollable(element)
+        ? element
+        : getScrollableElement(element.parentElement, ownerDocument);
+    };
+
+    const isElementScrollable = (element: HTMLElement) => {
+      const computedStyle = getComputedStyle(element);
+      return /(auto|scroll|overlay)/.test(
+        computedStyle.getPropertyValue('overflow') +
+          computedStyle.getPropertyValue('overflow-y'),
+      );
+    };
+
+    const virtualizer = useVirtualizer({
+      count: flatNodesList.length,
+      getScrollElement: () =>
+        getScrollableElement(
+          parentRef.current,
+          parentRef.current?.ownerDocument,
+        ),
+      estimateSize: () => 39,
```

Comment:
This number was taken from the height of the tree items that were used in the virtualized tree react story. 
![image](https://github.com/iTwin/iTwinUI/assets/141063311/7ce12634-0436-44a7-8919-a340a8eef638)
The basic tree items are a little shorter at 36 pixels, but the virtualization for this component is set up to support dynamic element sizing, so it will correct the size of the items when the tree component renders.

Reply:
I think that when the docs say within comfort, what they mean is that the largest size should be used that it makes reasonable sense for the item to be. In our case, the items could have theoretically infinite height, since our users can use the style prop to set the font size to be as large as they want. We can't really anticipate how large the user is going to set their font size, so there is no comfortable way to guess how large the estimate should be. I think it is better to estimate off of the default font size for now, and if there is a commonly used larger font size, we can adjust the calculations with that font size in mind. I have left the estimate at `39px` for now. Added a comment explaining where the number comes from in this [commit](https://github.com/iTwin/iTwinUI/pull/2091/commits/05060523af876477da6ec5e1038c7f00631ec760).